akka-actor/src/main/scala/akka/util/CachedSystemTimer.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.util
+
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.locks.LockSupport
+
+object CachedSystemTimer {
+  private var notAccuracyCounter = 0
+  private final val FORCE_UPDATE_THRESHOLD = 1000
+  import scala.concurrent.duration._
+  private final val ONE_MILLI_IN_NANOS: Int = 1.milli.toNanos.toInt
+  private final val cachedCurrentTimeMillisRef = new AtomicLong()
+  private final val updater = new Thread(new Runnable {
+    override def run(): Unit = {
+      while (true) {
+        val updated = System.currentTimeMillis()
+        cachedCurrentTimeMillisRef.set(updated)
+        LockSupport.parkNanos(ONE_MILLI_IN_NANOS)
+      }
+    }
+  })
+  updater.setDaemon(true)
+  updater.setName("CachedSystemTimer-Updater-Thread")
+  updater.start()
```

Comment:
How do you manage the lifecycle of this? For exmaple when running tests from sbt unforked this is going to end up leaking a timer thread per test run.

Better make it an extension, also - can't you use the system scheduler for the update rather than wasting a thread (which the scheduler already does), do you think the accuracy will suffer too much?

Reply:
Yes this should be an extension, if you think this is useful I will continue to go that way.
I extract this from our system.

Reply:
WDYT about the scheduler vs blocking thread part?